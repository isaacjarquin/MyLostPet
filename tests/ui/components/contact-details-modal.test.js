import "react-native"
import React from "react"
import ContactDetailsModal from "../../../components/contact-details-modal"

import { shallow } from "enzyme"
import toJSON from "enzyme-to-json"

describe("ContactDetailsModal", () => {
    const navigate = jest.fn()

    const props = {
        navigation: {
            navigate: navigate
        }
    }

    beforeEach(() => {
        wrapper = shallow(<ContactDetailsModal {...props} />)
    })

    afterEach(() => {
        navigate.mockReset()
    })

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    describe("_showModal", () => {
        it("shows modal", () => {
            wrapper.setState({ isModalVisible: false })
            wrapper.instance()._showModal()
            expect(wrapper.state().isModalVisible).toBe(true)
        })
    })

    describe("_hideModal", () => {
        it("hides modal", () => {
            wrapper.setState({ isModalVisible: true })
            wrapper.instance()._hideModal()
            expect(wrapper.state().isModalVisible).toBe(false)
        })
    })

    describe("name", () => {
        it("sets name", () => {
            wrapper.setState({ name: "", email: "test@gmail.com", phoneNumber: "034534535345", personalInformation: "personalInformation" })
            wrapper.instance().setValidations()
            expect(wrapper.state().name).toEqual({ "backgroundColor": "#FF6961", "icon": "cross", "iconType": "entypo", "validationFieldBorderColor": "red", "validationMessage": "El campo nombre es obligatorio", "validationMessageColor": "red" })
        })
    })

    describe("email", () => {
        it("sets email", () => {
            wrapper.setState({ name: "name", email: "test", phoneNumber: "034534535345", personalInformation: "personalInformation" })
            wrapper.instance().setValidations()
            expect(wrapper.state().email).toEqual({ "backgroundColor": "#FF6961", "icon": "cross", "iconType": "entypo", "validationFieldBorderColor": "red", "validationMessage": "Debes introducir un formato valido de email", "validationMessageColor": "red" })
        })
    })

    describe("phoneNumber", () => {
        it("sets phoneNumber", () => {
            wrapper.setState({ name: "name", email: "test@gmail.com", phoneNumber: "", personalInformation: "personalInformation" })
            wrapper.instance().setValidations()
            expect(wrapper.state().phoneNumber).toEqual({ "backgroundColor": "#FF6961", "icon": "cross", "iconType": "entypo", "validationFieldBorderColor": "red", "validationMessage": "Debes introducir un numero valido de telefono", "validationMessageColor": "red" })
        })
    })

    describe("personalInformation", () => {
        it("sets personalInformation", () => {
            wrapper.setState({ name: "name", email: "test@gmail.com", phoneNumber: "034534535345", personalInformation: "" })
            wrapper.instance().setValidations()
            expect(wrapper.state().personalInformation).toEqual({ "backgroundColor": "#FF6961", "icon": "cross", "iconType": "entypo", "validationFieldBorderColor": "red", "validationMessage": "El campo informacion personal es obligatorio", "validationMessageColor": "red" })
        })
    })
})
