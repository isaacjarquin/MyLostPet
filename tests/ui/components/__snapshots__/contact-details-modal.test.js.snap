// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F7",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#333333",
        "flexDirection": "row",
        "opacity": 0.8,
        "padding": 25,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 18,
          "marginLeft": "35%",
        }
      }
    >
      Contactar
    </Text>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
    />
    <View
      collapsable={undefined}
      onModalHide={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#333333",
          "borderTopLeftRadius": 3,
          "borderTopRightRadius": 3,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 37.5,
          "marginTop": 200.1,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <RCTScrollView>
        <View>
          <TextInput
            allowFontScaling={true}
            borderColor="grey"
            onChangeText={[Function]}
            placeholder="Nombre"
            placeholderTextColor="grey"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#333333",
                "borderWidth": 1,
                "color": "black",
                "fontSize": 16,
                "height": 60,
                "paddingLeft": 20,
                "paddingRight": 20,
              }
            }
            underlineColorAndroid="white"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            borderColor="grey"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="Correo"
            placeholderTextColor="grey"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#333333",
                "borderWidth": 1,
                "color": "black",
                "fontSize": 16,
                "height": 60,
                "paddingLeft": 20,
                "paddingRight": 20,
              }
            }
            underlineColorAndroid="white"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            borderColor="grey"
            keyboardType="phone-pad"
            onChangeText={[Function]}
            placeholder="Número de teléfono"
            placeholderTextColor="grey"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#333333",
                "borderWidth": 1,
                "color": "black",
                "fontSize": 16,
                "height": 60,
                "paddingLeft": 20,
                "paddingRight": 20,
              }
            }
            underlineColorAndroid="white"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            borderColor="grey"
            onChangeText={[Function]}
            placeholder="Información personal"
            placeholderTextColor="grey"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#333333",
                "borderWidth": 1,
                "color": "black",
                "fontSize": 16,
                "height": 60,
                "paddingLeft": 20,
                "paddingRight": 20,
              }
            }
            underlineColorAndroid="white"
            value=""
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "grey",
                "flexDirection": "row",
                "opacity": 1,
                "padding": 20,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/icons/send.png",
                }
              }
              style={
                Object {
                  "height": 25,
                  "marginLeft": "15%",
                  "marginRight": 10,
                  "width": 25,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Enviar mis datos
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </Modal>
</View>
`;
